pyls/plugins/symbols.py

```diff
@@ -9,18 +9,47 @@
 @hookimpl
 def pyls_document_symbols(config, document):
     all_scopes = config.plugin_settings('jedi_symbols').get('all_scopes', True)
-    definitions = document.jedi_names(all_scopes=all_scopes)
-    return [{
-        'name': d.name,
-        'kind': _kind(d),
-        'location': {'uri': document.uri, 'range': _range(d)}
-    } for d in definitions]
+    names = document.jedi_names(all_scopes=all_scopes)
+    return list(filter(None, [serialize_name(n, document.uri) for n in names]))
 
 
-def _range(d):
+def serialize_name(name, uri):
+    result = {}
+    parent = None
+
+    try:
+        # Jedi sometimes fails here.
+        parent = name.parent()
+    except:
+        pass
+
+    # - Only include global assignments in outline.
+    # - Don't include underscore assignments in outline.
+    # - Don't include function parameters in outline.
```

Comment:
Python language server shouldn't have opinions about this. The client can decide to filter out these symbols if it wishes.

Reply:
Happy to have a config option for this.
Though I don't know any other language server that displays parameters and assignments inside a function. So this is mostly to bring `pyls` inline with other language servers.

Displaying all function parameters and assignments becomes messy very fast and in my opinion mitigates most of the advantage of having a outline view.

Reply:
Other language servers behaviour is irrelevant in my opinion. The protocol is the source of truth here. 

However, since there is no SymbolKind for a function parameter, i agree it makes sense not to return them. Happy to filter those out. But underscore assignments and global assignments should (and could) be filtered by the client.

Reply:
👍 for not filtering out underscore assignments. It's just a thing I saw Nuclide do which felt neat.

> However, since there is no SymbolKind for a function parameter, i agree it makes sense not to return them.

I agree. That's how I got started doing all the filtering.
When only removing the function parameter from the outline view it feels strange when having a function like:
```python
def f(a):
    b = a
    return b
```
The outline view would show:
```bash
f
  a # only if also returning the parameter
  b
```
Which in my opinion is strange since `b` isn't a real property of `f`.

I think it is OK for the language server to make some language specific decisions what values to return.

Implementations in multi language editors like Atom or VSCode cannot easily make language agnostic decisions about displaying. So I think it's OK to provide some sensible defaults from the language server side.